src/librustc_mir/interpret/eval_context.rs

```diff
@@ -345,15 +345,19 @@ impl<'mir, 'tcx: 'mir, M: Machine<'mir, 'tcx>> InterpretCx<'mir, 'tcx, M> {
         local: mir::Local,
         layout: Option<TyLayout<'tcx>>,
     ) -> InterpResult<'tcx, TyLayout<'tcx>> {
-        match frame.locals[local].layout.get() {
+        // In trivial constants, we don't allocate any locals, so our `local` access may fail
+        // when it would otherwise be fine (e.g. when we're trying to get `mir::RETURN_PLACE`).
```

Comment:
why is this even happening? is this during the backtrace printing? do you have a backtrace of the ICE?

Reply:
It's trying to evaluate the uninhabited constant (is this something it shouldn't be doing?) — there's a back trace in https://github.com/rust-lang/rust/issues/61744.

Reply:
This is just for a sanity check with diverging functions. I believe that in https://github.com/rust-lang/rust/blob/24ddd1615419be89828fb5628e3c14af86c08b01/src/librustc_mir/interpret/terminator.rs#L393 we should just be using `self.frame().body.local_decls[local].ty.is_ty_uninhabited_from_all_modules()` and skip the entire machinery.

From my reading of the docs I think we should also rename `is_ty_uninhabited_from_all_modules` to `is_ty_uninhabited_from_any_module`

Reply:
Fixed.